card/mod/standard/set/all/path.rb

```diff
@@ -96,9 +100,31 @@ def path_extension opts
   end
 
   def path_query opts
+    opts = cast_path_opts opts
     opts.empty? ? "" : "?#{opts.to_param}"
   end
 
+  # normalizes certain path opts to specified data types
+  def cast_path_opts opts, cast_hash=CAST_PARAMS
+    return opts unless opts.is_a?(::Hash)
+    opts.each do |key, value|
+      next unless (cast_to = cast_hash[key])
+      opts[key] = cast_path_value value, cast_to
+    end
```

Comment:
negligible worse performance (more method calls) but much nicer to read:
 `opts.transform_values!(&:cast_path_value)` and do the conditional stuff in `cast_path_value`

Reply:
that won't work though. the cast hash depends on the key.

Reply:
right, ruby needs a `with_keys` method like the `with_index` method